### PR TITLE
Add org-level workspace class tests

### PR DESCRIPTION
**Pull request by mustard-mh**

### CODE REVIEW

File: tests/pages.ts [hunk 1]
```diff
@@ -79,21 +82,80 @@ export const workspaces = {
...
+            await page.locator('div').filter({ hasText: /^Should have one workspace class selected at least\.$/ }).isVisible();
+            return;
+        }
+        for (const id of AllWorkspaceClass) {
+            expect(await page.locator('#' + id).isChecked(), 'Class ' + id + ' should be').toBe(classes.includes(id));
+        }
+    }
+}
+
+export const sleep = (ms: number) => new Promise(res => setTimeout(res, ms))
```
Comment (akosyakov) [inline tests/pages.ts:134]: we should not use it, we shoulld never rely on timing

besides there is already `page.waitForTimeout`

File: tests/pages.ts [hunk 2]
```diff
@@ -79,21 +82,80 @@ export const workspaces = {
...
+        await orgSettings.goTo(page);
+        
+        await Promise.all(AllWorkspaceClass.filter(e => classes.includes(e)).map(e => orgSettings.checkWorkspaceClass(page, e, true)))
+        await Promise.all(AllWorkspaceClass.filter(e => !classes.includes(e)).map((e, i) => orgSettings.checkWorkspaceClass(page, e, false)))
+        if (classes.length === 0) {
+            await page.locator('div').filter({ hasText: /^Should have one workspace class selected at least\.$/ }).isVisible();
+            return;
+        }
+        for (const id of AllWorkspaceClass) {
+            expect(await page.locator('#' + id).isChecked(), 'Class ' + id + ' should be').toBe(classes.includes(id));
```
Comment (akosyakov) [inline tests/pages.ts:129]: we should use web assertions: https://playwright.dev/docs/best-practices#dont-use-manual-assertions they don't wait properly, it is a reason flakiness

File: tests/pages.ts [hunk 3]
```diff
@@ -79,21 +82,80 @@ export const workspaces = {
...
+            await checkBox.click();
+        }
+    },
+    setAllowedWorkspaceClasses: async (page: Page, classes: WorkspaceClass[]) => {
+        await orgSettings.goTo(page);
+        
+        await Promise.all(AllWorkspaceClass.filter(e => classes.includes(e)).map(e => orgSettings.checkWorkspaceClass(page, e, true)))
+        await Promise.all(AllWorkspaceClass.filter(e => !classes.includes(e)).map((e, i) => orgSettings.checkWorkspaceClass(page, e, false)))
+        if (classes.length === 0) {
+            await page.locator('div').filter({ hasText: /^Should have one workspace class selected at least\.$/ }).isVisible();
```
Comment (akosyakov) [inline tests/pages.ts:125]: don't use such locator

use `expect(page.getText('/^Should have one workspace class selected at least\.$/)).toBeVisibile()`

it relies on rendered text and waits properly

File: tests/pages.ts [hunk 4]
```diff
@@ -79,21 +82,80 @@ export const workspaces = {
...
+    setWorkspaceShared: async (page: Page, shared: boolean) => {
+        await page.getByLabel('Workspace SharingAllow').click();
+        expect(await page.getByLabel('Workspace SharingAllow').isChecked()).toBe(shared);
+    },
+    checkWorkspaceClass: async (page: Page, id: WorkspaceClass, checked: boolean) => {
+        const isChecked = await page.locator('#' + id).isChecked();
+        if ((isChecked && checked) || (!isChecked && !checked)) {
+            return;
+        }
+        const checkBox = page.locator("#" + id)
```
Comment (akosyakov) [inline tests/pages.ts:112]: maybe simply `click({force: true})` is enough

File: tests/pages.ts (hunk 1 above)
Comment (mustard-mh) [reply tests/pages.ts:134]: I add sleep to make sure I can see the browser render 🤣, test goes too fast, I can’t see it ()

Going to remove it

Comment (akosyakov) [reply tests/pages.ts:134]: you can debug the, or add slowmo https://playwright.dev/docs/debug#headed-mode to profile in the configuration

**Comment by mustard-mh** [comment]
Not sure why but expect toBeVisible can't find the checkbox I wanted, although it does exist

Going to push it 🙏 
image

**Comment by mustard-mh** [comment]
@akosyakov Updated, looks stable now

**Review by akosyakov — APPROVED** [review]

**Comment by mustard-mh** [comment]
https://github.com/gitpod-io/dashboard-playwright-tests/pull/4

**Pull request closed without merging**